### ISSUE TITLE
QC ML quantbooks : le nom promet un moteur (LSTM, XGBoost) que le notebook n'invoque pas

### ISSUE THREAD

**Issue by jsboige** [jsboige/CoursIA#8757]
Constate en reviewant #8756, mais **le pattern pre-existe sur `main`** — il n'est pas du fait de cette PR, que j'ai mergee (`68de4f59e`). Je l'ai sorti de son scope pour le traiter proprement ici.

## Le constat

`MyIA.AI.Notebooks/QuantConnect/projects/ML-DeepLearning/quantbook.ipynb` s'appelle **ML-DeepLearning**. Le modele qu'il entraine reellement est un **`Ridge` / `RandomForest` scikit-learn**. Le code Keras est present, mais **en commentaire**.

Compte des occurrences, identique sur `main` et sur la PR (donc non introduit par la re-exec) :

```
Ridge            7        keras            4
RandomForest     1        tensorflow       7
                          LSTM            20
```

Les 20 mentions de `LSTM` et les 7 de `tensorflow` decrivent un modele que le notebook **n'execute pas**. Les sorties committees sont celles du proxy sklearn.

Le manifeste `lean_universes.manifest.json` le dit d'ailleurs franchement, ce qui est a porter au credit de l'auteur :

> « the notebook trains an sklearn Ridge/RandomForest proxy as a stand-in for a real LSTM (documented in-cell, real Keras code shown in a comment) »

## Pourquoi c'est un probleme de fond

C'est le cas (A) de Prong A de [`sota-not-workaround.md`](../blob/main/.claude/rules/sota-not-workaround.md) : **reimplementation jouet a la place de la lib reelle**. La particularite ici est que le nom du notebook *est* la promesse — un lecteur qui ouvre `ML-DeepLearning` pour apprendre le deep learning applique au trading lit du code Keras commente et voit tourner une regression lineaire regularisee.

Ridge et un LSTM ne different pas par le reglage : ils different par ce qu'ils peuvent apprendre. Un Ridge sur features tabulaires ne modelise aucune dependance temporelle longue — precisement la capacite qui justifie un LSTM sur des series de prix. Le notebook ne peut donc pas *montrer* ce qu'il enseigne.

A noter que le notebook est honnete en cellule (le proxy est documente), ce qui le place dans « perte documentee » plutot que « perte par complaisance ». Le defaut n'est pas le mensonge, c'est le plafond.

## Ce qu'il faut trancher (je ne prejuge pas du verdict)

**Je ne sais pas** si TensorFlow/Keras est invocable dans le runtime ou tourne ce quantbook — c'est la premiere chose a etablir, et elle decide de tout le reste :

1. **`import tensorflow` reussit dans l'environnement de re-exec** → verdict **RECOVERABLE-LOCAL** : decommenter le vrai modele, entrainer, re-executer, committer les vraies sorties. Le proxy disparait.
2. **TF absent mais installable** (`pip install tensorflow` sur la machine de la lane, regle F) → **RECOVERABLE-LOCAL** aussi : on installe, on ne contourne pas.
3. **TF genuinement indisponible** (incompatibilite runtime QC / poids GPU / cout de build prohibitif) → **INTRINSIC documente** : le proxy reste, mais le notebook doit dire **en tete**, pas en note de bas de cellule, qu'il substitue un modele lineaire, **pourquoi**, et ce que le LSTM apporterait — avec le chiffre ou l'erreur qui motive la substitution.

Ordre de travail : etablir 1/2/3 **par la mesure** (lancer l'import, rapporter la sortie), pas par estimation. C'est le meme protocole que #8749 pour `native_decide` — on mesure avant de declarer intractable.

## Verification

Soit le notebook entraine un vrai modele Keras et committe ses sorties, soit il porte en tete une justification chiffree de la substitution.

Meme famille que #8750 (cellules de rejeu `10_LocalLlama`) : un notebook demonstratif dont le coeur pedagogique n'est plus produit par le code qu'on lit.

See #8734, #3801, #8052.

**Comment by jsboige** [comment]
**Deuxieme instance mesuree — meme classe, meme famille : `ML-XGBoost` n'utilise pas XGBoost.**

Trouve en reviewant #8760 (que j'ai mergee, `19dfe2c6f` — elle est strictement benefique : le notebook passe de 1/11 casse a 11/11 fonctionnel). Le defaut est **pre-existant**, pas introduit par cette PR. Je le sors de son scope pour le traiter ici, comme j'avais fait pour ML-DeepLearning.

## La mesure (firsthand, `main` avant et branche apres)

```
                          main    PR #8760
xgboost (mentions)          5         6
XGBRegressor                0         0
XGBClassifier               0         0
GradientBoostingRegressor   7         8
```

`XGBRegressor` et `XGBClassifier` sont a **zero des deux cotes**. Le notebook mentionne « xgboost » 5 fois en prose et entraine un `GradientBoostingRegressor` de scikit-learn. Le body de #8760 le dit franchement — « GradientBoostingRegressor **stand-in** » — ce qui place l'instance en *perte documentee*, pas en *perte par complaisance*.

## Pourquoi c'est la meme issue et pas une nouvelle

C'est exactement le motif deja decrit ici pour `ML-DeepLearning` (Ridge/RandomForest la ou le nom promet un LSTM) : **le nom du notebook EST la promesse**, et le moteur nomme n'est jamais invoque. Deux instances, une seule classe de defaut, un seul protocole de resolution. J'elargis donc cette issue au lieu d'en ouvrir une quasi-jumelle.

| Notebook | Promesse du nom | Ce qui tourne reellement | Lib a sonder |
|---|---|---|---|
| `ML-DeepLearning` | LSTM / deep learning | `Ridge` + `RandomForest` (Keras en commentaire) | `tensorflow` |
| `ML-XGBoost` | XGBoost | `GradientBoostingRegressor` (sklearn) | `xgboost` |

## Le protocole ne change pas — mesurer avant de declarer

Pour **chacune** des deux libs, dans le runtime ou le quantbook s'execute reellement (conteneur `lean research`, celui que po-2024 a chaud depuis #8760) :

1. `import tensorflow` / `import xgboost` **reussit** -> **RECOVERABLE-LOCAL** : brancher le vrai modele, re-executer, committer les vraies sorties. Le proxy disparait.
2. **Absent mais installable** (regle F) -> **RECOVERABLE-LOCAL** aussi : on installe, on ne contourne pas.
3. **Genuinement indisponible** (incompat runtime QC, poids GPU, cout de build prohibitif) -> **INTRINSIC documente** : le proxy reste, mais le notebook dit **en tete** qu'il substitue, **pourquoi**, et ce que le vrai moteur apporterait — avec le chiffre ou l'erreur qui motive la substitution.

**Rapporter la sortie de l'import, pas une estimation.** C'est le meme protocole que #8749 pour `native_decide` : on mesure avant de declarer intractable.

Note de cadrage : `xgboost` est nettement plus leger que `tensorflow` (pas de CUDA requis, wheel CPU standard). Il est donc plausible que les deux instances tombent dans des branches differentes — c'est precisement pourquoi la mesure se fait **lib par lib**, pas en bloc.

## Verification

Pour chaque notebook : soit il entraine le moteur que son nom annonce et committe ses vraies sorties, soit il porte en tete une justification **chiffree** de la substitution.

See #8760, #3801, #8052.

**Comment by myia-ai-01** [comment]
## Fermee — les deux moteurs promis sont desormais invoques, mesure a l'appui

L'acceptance de ce ticket etait : *« Soit le notebook entraine un vrai modele Keras et committe ses sorties, soit il porte en tete une justification chiffree de la substitution. »* **C'est la premiere branche qui est atteinte, pour les deux libs.**

| Lib | PR | Merge | Ce qui remplace quoi |
|---|---|---|---|
| XGBoost | #8767 | `a56020acd` | `XGBRegressor` remplace le stand-in `GradientBoostingRegressor` |
| TensorFlow/Keras LSTM | #8770 | `66ff07c81` | `Sequential[LSTM, Dropout, Dense]` remplace le proxy `Ridge` qui aplatissait les sequences 3D |

### Le point d'ordre du ticket a ete respecte

Le ticket demandait d'etablir 1/2/3 **par la mesure**, pas par estimation. po-2024 a sonde le runtime reel — `quantconnect/research:latest`, l'image ou ces quantbooks s'executent :

```
import xgboost      -> 3.0.5     (exit 0)
tensorflow 2.19.1 | keras 3.13.0 | torch 2.8.0
```

**Les deux libs etaient installees tout du long.** Verdict **RECOVERABLE-LOCAL** pour les deux — aucun des cas 2 ou 3 n'a eu a etre invoque. Le plafond que decrivait ce ticket n'etait pas technique, il etait d'inattention.

### Verifie firsthand avant fermeture (pas depuis les bodies)

Sur `ML-DeepLearning/quantbook.ipynb` a l'etat merge : **12/12 cellules code**, `execution_count` **1..12 sequentiel**, **0 erreur**, **0 cellule sans sortie**. 4 `.fit` / 2 `.compile` reels. La banniere d'initialisation cuDNN/cuBLAS presente dans les sorties est la preuve que TensorFlow a effectivement charge en container — le silence des epochs s'explique par `verbose=0`, pas par une absence d'entrainement. Du `Ridge` il ne reste qu'un commentaire qui **documente son retrait**.

Mesures committees : walk-forward `MSE 0.000727`, `Direction Accuracy 52.42%`.

### Deux reserves tracees ailleurs, qui ne bloquent pas cette fermeture

1. **Prong-B sur ML-XGBoost (#3801)** : quatre modeles a R² ≈ 0 ne discriminent pas — la comparaison n'exerce pas la capacite propre de XGBoost. Grain distinct.
2. **Prong-B sur ML-DeepLearning** : 15 epochs (principal) / 5 par tranche walk-forward, borne assume pour le temps d'execution pedagogique. La regle carve-out explicitement le « temps de traitement raisonnable », donc ce n'est pas un manquement — mais une direction accuracy de **52.42 %** est a un cheveu du pile-ou-face, et le rendement de backtest de **+30,18 %** est publie **sans baseline buy-and-hold** en regard. Ce n'est pas une revendication de BEATS (le body n'en fait pas), donc §C ne se declenche pas ; c'est le chiffre qui **pourrait** se lire comme un edge s'il n'etait pas encadre. A reprendre dans le grain Prong-B.

Ces deux reserves portent sur la **richesse du probleme**, pas sur l'outil : Prong-A est tenu, c'etait l'objet de ce ticket.

See #8767, #8770, #3801.
